### ISSUE TITLE
Code only implements BCE loss (classification) — MSE loss path for regression described in paper is missing

### ISSUE THREAD

**Issue by talha1369** [szhang1112/scPRS#10]
## Description

The paper Methods state:

> "For the regression task, the mean squared error was used as the loss function instead of BCE."

And the simulation experiments (Extended Data Fig. 1, Supplementary Fig. 1) demonstrate a regression task: predicting simulated monocyte counts as a continuous value, evaluated with Pearson correlation.

However, `main.py` only implements BCE loss (line 62):

```python
loss = F.binary_cross_entropy_with_logits(out, yy)
```

There is:
- No MSE code path
- No flag to switch between classification and regression (e.g., `--task regression`)
- The minority-class upsampling logic (line 55) assumes binary labels (`y_id[0]==1`)

A user trying to apply scPRS to a continuous phenotype — as described in the paper and demonstrated in the simulations — would need to substantially modify the training loop.

## Suggested addition

A `--task` argument that switches between BCE (classification) and MSE (regression):

```python
parser.add_argument("--task", type=str, default="classification", choices=["classification", "regression"])
```

With the training loop using `F.mse_loss(out, yy)` for regression and skipping the minority upsampling.

## Question

Could you add the regression code path, or confirm whether a separate script was used for the regression experiments in the paper?
